name: "Run E2E chaos test"

on:
  # TODO: Delete it later.
  push:
    branches:
      - "add-dump-action"

jobs:
  dump_contexts_to_log:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: ./.github/actions/dump-context

  agent-failure:
    name: "E2E chaos test (Agent failure: to test insert/search works even if one of the agents is failing)"
    runs-on: ubuntu-latest
    timeout-minutes: 60
    container:
      image: ghcr.io/vdaas/vald/vald-ci-container:nightly
    steps:
      - uses: actions/checkout@v3
        with:
          path: ${{ github.workspace }}
          set-safe-directory: true
      # - uses: ./.github/actions/setup-go
      # - uses: ./.github/actions/setup-helm
      - uses: rinx/setup-k3d@v0.0.4
        with:
          version: latest
          name: vald
          agents: 3
          options: "--image docker.io/rancher/k3s:latest"
      - name: Deploy Chaos Mesh
        uses: ./.github/actions/deploy-chaos-mesh
        with:
          helm_extra_options: "--set podChaos.failure.enabled=true"
